### ISSUE TITLE
Fix punycode deprecation warning since Node 21

### ISSUE THREAD

**Issue by ehmicky** [release-it/release-it#1051]
The following warning is currently printed when using Node 21:

```
(node:17753) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
```

This is coming from https://github.com/jsdom/whatwg-url/blob/d34854a7af6ed1204f55d7da7761497bad350a7b/src/url-state-machine.js#L2 which is installed through:

```
└─┬ release-it@16.2.1
  └─┬ @octokit/rest@19.0.13
    └─┬ @octokit/core@4.2.4
      └─┬ @octokit/request@6.2.8
        └─┬ node-fetch@2.6.12
          └── whatwg-url@5.0.0
```

The latest version of `whatwg-url` does not print this warning anymore. However, this requires upgrading `@octokit/rest` from v19 to [v20](https://github.com/octokit/rest.js/releases/tag/v20.0.0). Doing so would drop support for Node 16 (which is not officially supported anymore by Node.js).

This would also remove the `agent` feature, which is currently used by the `github.proxy` option.

https://github.com/release-it/release-it/blob/9e99e2b2c7787f874e29e22541bdb317015dd61d/lib/plugin/github/GitHub.js#L181-L183

The original PR that removed that feature is https://github.com/octokit/request.js/pull/599. Instead, a [custom `fetch`](https://github.com/octokit/request.js#set-a-custom-agent-to-your-requests) should be passed instead. If support for Node `<18.13.0` is removed, the global `fetch()` method can be used without printing an experimental warning. Under the hood, it uses [`undici.fetch()`](https://github.com/nodejs/undici#undicifetchinput-init-promise). An `Agent` can be passed, but I am unclear whether it must be an [`undici` `Agent`](https://github.com/nodejs/undici#undicifetchinput-init-promise) (which would require adding `undici` as a dependency) or a core Node.js `http` `Agent`.

**Comment by webpro** [comment]
Thanks for the detailed breakdown @ehmicky, I'll look into it.

**Comment by webpro** [comment]
This should've been fixed by ebc2754ef0609e3597bd138cc127d171d8a8850a and releasing v17.
